e2etests/orchestration/common/common.go

```diff
@@ -132,6 +133,29 @@ func (h *DockerHelper) RunContainer(img string, hostPort int) (string, error) {
 	return createRes.ID, nil
 }
 
+func (h *DockerHelper) getIpAddr(id string) (string, error) {
+	ctx := context.TODO()
+	listRes, err := h.client.ContainerList(ctx, container.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+	var found *types.Container
+	for _, container := range listRes {
+		if container.ID == id {
+			found = &container
+			break
+		}
+	}
+	if found == nil {
+		return "", fmt.Errorf("container not found: %q", id)
+	}
```

Comment:
Instead of looping over all containers here, could the client have a `ContainerById` method instead?

Reply:
Good call, used [ContainerInspect](https://github.com/docker/docker/blob/v28.0.1/client/container_inspect.go#L14).